src/transport/SessionHandle.cpp

```diff
@@ -39,4 +39,22 @@ const PeerAddress * SessionHandle::GetPeerAddress(SessionManager * sessionManage
     return &GetUnauthenticatedSession()->GetPeerAddress();
 }
 
+CHIP_ERROR SessionHandle::GetMRPIntervals(SessionManager * sessionManager, uint32_t & mrpIdleInterval, uint32_t & mrpActiveInterval)
+{
+    if (IsSecure())
+    {
+        SecureSession * secureSession = sessionManager->GetSecureSession(*this);
+        if (secureSession == nullptr)
+        {
+            return CHIP_ERROR_INVALID_ARGUMENT;
+        }
+        secureSession->GetMRPIntervals(mrpIdleInterval, mrpActiveInterval);
+
+        return CHIP_NO_ERROR;
+    }
+    GetUnauthenticatedSession()->GetMRPIntervals(mrpIdleInterval, mrpActiveInterval);
```

Comment:
nit: I'm probably pedantic - but why not to make a new line always after "}". It makes code more readable.

Reply:
Done